containers/modal/edit-expiry-modal-container.tsx

```diff
@@ -59,6 +97,120 @@ const EditExpiryModalContainer = ({
     setEpochs(expiryEpochIndex)
   }, [expiryEpochIndex, position, setEpochs])
 
+  const { data: adjustPositionSimulateData } = useQuery(
+    [
+      'edit-expiry-adjust-position-simulate',
+      selectedChain,
+      position,
+      epochs,
+      expiryEpochIndex,
+      interest,
+      payable,
+      refund,
+      refundable,
+      maxLoanableAmountExcludingCouponFee,
+    ],
+    async () => {
+      const newToEpoch = position.toEpoch.id + epochs - expiryEpochIndex
+      const marketsBeforeEditExpiry = (
+        await fetchMarketsByQuoteTokenAddress(
+          selectedChain.id,
+          position.substitute.address,
+        )
+      ).filter(
+        (market) => market.epoch <= Math.max(newToEpoch, position.toEpoch.id),
+      )
+
+      // extend or shorten loan duration
+      if (position.toEpoch.id < newToEpoch && payable) {
+        // extend
+        for (let i = 0; i < marketsBeforeEditExpiry.length; i++) {
+          if (position.toEpoch.id < marketsBeforeEditExpiry[i].epoch) {
+            ;({ market: marketsBeforeEditExpiry[i] } = marketsBeforeEditExpiry[
+              i
+            ].take(
+              marketsBeforeEditExpiry[i].quoteToken.address,
+              position.amount,
+            ))
+          }
+        }
+        const marketsAfterEditExpiry = marketsBeforeEditExpiry.slice()
+        const calculateCouponsToBorrowResult = calculateCouponsToBorrow(
+          position.substitute,
+          marketsAfterEditExpiry,
+          maxLoanableAmountExcludingCouponFee,
+          interest,
+        )
+        return {
+          positionAmountDelta:
+            interest + calculateCouponsToBorrowResult.interest,
+          enoughCoupon: interest <= calculateCouponsToBorrowResult.available,
+          enoughCollateral:
+            interest <=
+            maxLoanableAmountExcludingCouponFee -
+              calculateCouponsToBorrowResult.maxInterest,
+        }
+      } else if (newToEpoch < position.toEpoch.id && refundable) {
```

Comment:
```suggestion
      } else if (newToEpoch < position.toEpoch.id && refundable) {
```
p2; remove `refundable`